src/App.js

```diff
@@ -9,94 +9,97 @@ class App extends Component {
   constructor() {
     super();
     this.state = {
-      areGamesLoaded: false,
-      areGenresLoaded: false,
-      error: null,
+      errors: null,
       games: null,
       genres: null,
       popUpInfo: null,
-      popUpGenreID: null,
+      popUpGenre: null,
       searching: '',
       filteredGames: []
     }
   }
+  getData = (request) => {
+    const url = 'https://whateverly-datasets.herokuapp.com/api/v1/';
 
-  componentDidMount() {
-    fetch('https://whateverly-datasets.herokuapp.com/api/v1/games')
+    fetch(`${url + request}`)
       .then(data => data.json())
       .then(result => this.setState({
-        areGamesLoaded: true,
-        games: result.games
+        [request]: result[request]
       }))
-      .catch(error => {
+      .catch(errors => {
         this.setState({
-          error
+          errors
         })
       })
+  }
 
-
-    fetch('https://whateverly-datasets.herokuapp.com/api/v1/genres')
-      .then(data => data.json())
-      .then(result => this.setState({
-        areGenresLoaded: true,
-        genres: result.genres
-      }))
-      .catch(error => {
-        this.setState({
-          error
-        })
-      })
+  componentDidMount() {
+    this.getData('games');
+    this.getData('genres');
   }
 
-  createPopUp = (event, genreID) => {
+  createPopUp = (event) => {
```

Comment:
I am up in the air on this change. 
### PROS:
A plus was taking out the second parameter, secondly we now have a generic method we can pass to popup everywhere. Previously we nest a call to this method inside an anonymous function. This made it difficult to refactor and reuse the popup component.

### CONS:
The createPopUp method has more functionality as well as increased complexity. Not sure if the trade-off is worth it to my teammates, awaiting feedback.

Reply:
I agree. Allowing more reusable code is worth the increased complexity in this case.